credentials/tls.go

```diff
@@ -48,6 +48,17 @@ func (t TLSInfo) AuthType() string {
 	return "tls"
 }
 
+func (t TLSInfo) ValidateAuthority(authority string) error {
+	var err error
+	for _, c := range t.State.PeerCertificates {
+		err = c.VerifyHostname(authority)
```

Comment:
can you link from where we took this verification logic? or how we are testing?

Reply:
okay. I was asking if there is any place in our library we have similar way to verify certs

Reply:
I dont think so